DMCompiler/DM/DMProc.cs

```diff
@@ -382,9 +382,10 @@ public void Pop()
         }
 
         public void PushArguments(int argumentCount, DreamProcOpcodeParameterType[] parameterTypes = null, string[] parameterNames = null) {
-            ShrinkStack(argumentCount - 1); //Pops argumentCount, pushes 1
+            ShrinkStack(argumentCount - 2); //Pops argumentCount & parameterNames.Length, pushes 1
```

Comment:
Double check me on this, I don't know the opcodes super well.

Reply:
This isn't correct. `WriteInt()` writes an operand for the opcode and doesn't affect the stack.